Author: Reto Gantenbein <reto.gantenbein@linuxmonk.ch>
Date:   Tue May 8 17:57:28 2017 +0200

    Set X-Forwarded headers when accessed via HTTPS

    The Apache httpd of the OMD site uses various
    rewrites for accessing the individual applications.
    Define the necessary headers so this can work
    properly when the proxy httpd is configured for
    HTTPS.

    Note: If the HTTP(S) port is not 80(443) this won't
    work correctly. Attempts to use the httpd variable
    %{SERVER_PORT} in in the X-Forwarded-Port header
    failed because it seemed to be undefined "(null)".

    v2: - Guard 'RequestHeader' statements with
          <IfModule> to avoid errors if mod_headers
          is not (yet) enabled

--- /omd/versions/default/skel/etc/apache/apache-own.conf.orig  2016-05-13 19:19:07.000000000 +0200
+++ /omd/versions/default/skel/etc/apache/apache-own.conf       2017-05-08 17:56:25.342383031 +0200
@@ -11,6 +11,12 @@
   ProxyRequests Off
   ProxyPreserveHost On
 
+  # Indicate when the site was accessed via HTTPS
+  <IfModule mod_headers.c>
+  RequestHeader set X-Forwarded-Proto https env=HTTPS
+  RequestHeader set X-Forwarded-Port 443 env=HTTPS
+  </IfModule>
+
   # Include file created by 'omd config', which 
   # sets the TCP port of the site local webserver
   Include ###ROOT###/etc/apache/proxy-port.conf
